src/apps/FEMSWEEP.cpp

```diff
@@ -0,0 +1,146 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-24, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "FEMSWEEP.hpp"
+#include "FEMSWEEP_DATA.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#include "common/DataUtils.hpp"
+
+#include <algorithm>
+#include <cmath>
+
+namespace rajaperf
+{
+namespace apps
+{
+
+
+FEMSWEEP::FEMSWEEP(const RunParams& params)
+  : KernelBase(rajaperf::Apps_FEMSWEEP, params)
+{
+  m_ne = 15 * 15 * 15;
+  m_na = 72;
+  m_ng = 128;
+
+  setDefaultProblemSize(ND * m_ne * m_ng * m_na);
+  setDefaultReps(1);
+
+  m_Blen = ND * m_ne * m_na;
+  m_Alen = ND * ND * m_ne * m_na;
+  // 9450 is a property of the mesh. Will need to derive this when mesh generator is available.
+  m_Flen = FDS * FDS * 2 * 9450 * m_na;
+  m_Sglen = m_ne * m_ng;
+  m_M0len = ND * ND * m_ne;
+  m_Xlen = ND * m_ne * m_ng * m_na;
+
+  setActualProblemSize( m_Xlen );
+
+  setItsPerRep(1);
+  setKernelsPerRep(1);
+  // using total data size instead of writes and reads
+  setBytesReadPerRep( 1*sizeof(Real_type) * m_Blen +
+                      1*sizeof(Real_type) * m_Alen +
+                      1*sizeof(Real_type) * m_Flen +
+                      1*sizeof(Real_type) * m_Sglen +
+                      1*sizeof(Real_type) * m_M0len +
+                      1*sizeof(Real_type) * m_Xlen );
+  setBytesWrittenPerRep( 1*sizeof(Real_type) * m_Xlen );
+  setBytesAtomicModifyWrittenPerRep( 0 );
+
+  // This is an estimate of the upper bound FLOPs.
+  setFLOPsPerRep( (ND * ND * (ND-1) * 3 * 2 +       // L & U formation
+                  ND * (ND-1) * 3 +                 // forward substitution
+                  ND * (ND-1) * 3 +                 // backward substitution
+                  NLF * FDS - pow(m_ne, 2/3) * 6) * // coupling between sides of faces
```

Comment:
Can you calculate this from the number of elements per side so we avoid floating point?